ApiTests. remove objects from space.

ScharfViktor opened the pull request:
- added test "create subfolder"
- added the ability to search objects in subfolders of the space 
- added tests "deleting object from space" and "deleting space"
- created bug: https://github.com/owncloud/ocis/issues/3031

phil-davis commented on tests/acceptance/features/apiSpaces/uploadSpaces.feature:
```diff
@@ -11,20 +11,32 @@ Feature: Upload files into a space
       | mainFolder        |
 
+
+  Scenario: A user can create a subfolders in a Space via the Graph API
```
```suggestion
  Scenario: A user can create subfolders in a Space via the Graph API
```

phil-davis commented on tests/acceptance/features/apiSpaces/uploadSpaces.feature:
```diff
@@ -11,20 +11,32 @@ Feature: Upload files into a space
+      | mainFolder |
+    And for user "Alice" folder "mainFolder/subFolder1/" of the space "Subfolders in Project" should contain these entries:
+      | subFolder2 |
+
```
You could add a 2nd blank line here, to be the same as the other scenarios.

phil-davis commented on tests/acceptance/features/apiSpaces/removeSpaces.feature:
```diff
@@ -0,0 +1,181 @@
+      | quota@@@used | 0                  |
+
+
+  Scenario: A editor can delete a empty Space via the webDav API
```
I am a bit surprised by this. But maybe this is the requirement?
I was thinking that spaces would have at least 3 roles like:
- Viewer - read-only of everything in the space
- Editor - can create/change/delete the resources (files and folders) in the space
- Manager - can manage access to the space (grant and revoke roles to users...) and can delete the whole space.

ScharfViktor replied:
AC in https://jira.owncloud.com/browse/OCIS-2063
viewer has no right to delete

ScharfViktor replied:
You are right, the editor should not be able to delete space.
https://jira.owncloud.com/browse/OCIS-2043
I will collect more information tomorrow and change some cases

phil-davis commented on tests/acceptance/features/apiSpaces/removeSpaces.feature:
```diff
@@ -0,0 +1,181 @@
+      | text.txt |
+
+
+  Scenario: A viewer cannot delete a empty Space via the webDav API
```
```suggestion
  Scenario: A viewer cannot delete an empty Space via the webDav API
```

phil-davis commented on tests/acceptance/features/apiSpaces/removeSpaces.feature:
```diff
@@ -0,0 +1,181 @@
+    Then the json responded should not contain a space with name "Editor deletes empty space"
+
+
+  Scenario: A editor can delete a Space containing folders and files via the webDav API
```
```suggestion
  Scenario: An editor can delete a Space containing folders and files via the webDav API
```

phil-davis commented on tests/acceptance/features/apiSpaces/removeSpaces.feature:
```diff
@@ -0,0 +1,181 @@
+      | quota@@@used | 0                  |
+
+
+  Scenario: A editor can delete a empty Space via the webDav API
```
```suggestion
  Scenario: An editor can delete an empty Space via the webDav API
```

phil-davis commented on tests/acceptance/features/apiSpaces/removeSpaces.feature:
```diff
@@ -0,0 +1,181 @@
+      | subFolderForDeleting |
+
+
+  Scenario: A editor can delete a file in a Space via the webDav API
```
```suggestion
  Scenario: An editor can delete a file in a Space via the webDav API
```

phil-davis commented on tests/acceptance/features/apiSpaces/removeSpaces.feature:
```diff
@@ -0,0 +1,181 @@
+      | folderForDeleting |
+
+
+  Scenario: A editor can delete a subfolder in a Space via the webDav API
```
```suggestion
  Scenario: An editor can delete a subfolder in a Space via the webDav API
```

phil-davis commented on tests/acceptance/features/apiSpaces/removeSpaces.feature:
```diff
@@ -0,0 +1,181 @@
+
+#   editor role
+
+  Scenario: A editor can delete a folder with some subfolders in a Space via the webDav API
```
```suggestion
  Scenario: An editor can delete a folder with some subfolders in a Space via the webDav API
```

phil-davis commented on tests/acceptance/features/apiSpaces/removeSpaces.feature:
```diff
@@ -0,0 +1,181 @@
+    Then the json responded should not contain a space with name "Owner deletes empty space"
+
+
+  Scenario: A owner can delete a Space containing folders and files via the webDav API
```
```suggestion
  Scenario: An owner can delete a Space containing folders and files via the webDav API
```

phil-davis commented on tests/acceptance/features/apiSpaces/removeSpaces.feature:
```diff
@@ -0,0 +1,181 @@
+      | quota@@@used | 0                 |
+
+
+  Scenario: A owner can delete a empty Space via the webDav API
```
```suggestion
  Scenario: An owner can delete an empty Space via the webDav API
```

phil-davis commented on tests/acceptance/features/apiSpaces/removeSpaces.feature:
```diff
@@ -0,0 +1,181 @@
+      | subFolderForDeleting |
+
+
+  Scenario: A owner can delete a file in a Space via the webDav API
```
```suggestion
  Scenario: An owner can delete a file in a Space via the webDav API
```

phil-davis commented on tests/acceptance/features/apiSpaces/removeSpaces.feature:
```diff
@@ -0,0 +1,181 @@
+      | folderForDeleting |
+
+
+  Scenario: A owner can delete a subfolder in a Space via the webDav API
```
```suggestion
  Scenario: An owner can delete a subfolder in a Space via the webDav API
```

phil-davis commented on tests/acceptance/features/apiSpaces/removeSpaces.feature:
```diff
@@ -0,0 +1,181 @@
+
+#   owner of space
+
+  Scenario: A owner can delete a folder with some subfolders in a Space via the webDav API
```
```suggestion
  Scenario: An owner can delete a folder with some subfolders in a Space via the webDav API
```

phil-davis reviewed (commented):
LGTM - mostly just the English word "a" and "an" :)

individual-it commented on tests/acceptance/features/apiSpaces/removeSpaces.feature:
```diff
@@ -0,0 +1,181 @@
+    And user "Brian" has been created with default attributes and without skeleton files
+    And the administrator has given "Alice" the role "Admin" using the settings api
+
+#   owner of space
```
what is this line about?

ScharfViktor replied:
I separated tests by permissions: owner, editor, viewer

ScharfViktor commented:
Paused. I am waiting for permissions change from @C0rby.
for disabling/removing space we should use another method: `curl -k -s -u admin:admin -X DELETE 'https://localhost:9200/graph/v1.0/drives/space_uuid'`
Tests will be changed

C0rby commented:
The current ocis master now contain the permission changes.
Disabling and deleting a space should now work in ocis master.

ScharfViktor commented:
I changed tests and bug report https://github.com/owncloud/ocis/issues/3031. PR is ready for review

I have added 3 tests for expected failures. In the bug report I described what was wrong in my opinion

individual-it reviewed (approved):
Tests look good to me, but we should discuss if adding the WebDAV tests is necessary, currently there is a lot of work going on to add "spaces" as a variant to the existing WebDAV tests. That should cover most WebDAV cases I think
see https://github.com/owncloud/ocis/issues/2854
